src/software/ai/hl/stp/tactic/pass_defender/pass_defender_fsm.cpp

```diff
@@ -105,3 +106,62 @@ void PassDefenderFSM::interceptBall(const Update& event)
         TbotsProto::DribblerMode::MAX_FORCE, TbotsProto::BallCollisionType::ALLOW,
         AutoChipOrKick{AutoChipOrKickMode::OFF, 0}));
 }
+
+bool PassDefenderFSM::ballNearbyWithoutThreat(const Update& event)
+{
+    Point robot_position = event.common.robot.position();
+    Point ball_position  = event.common.world_ptr->ball().position();
+
+    std::optional<Robot> nearest_friendly_to_ball =
+        event.common.world_ptr->friendlyTeam().getNearestRobot(ball_position);
+    std::optional<Robot> nearest_enemy_to_ball =
+        event.common.world_ptr->enemyTeam().getNearestRobot(ball_position);
+    if (event.control_params.ball_steal_mode == TbotsProto::BallStealMode::IGNORE)
+    {
+        // Do nothing if stealing is disabled
+        return false;
+    }
+    else if (nearest_friendly_to_ball.has_value() &&
+             event.common.robot.id() != nearest_friendly_to_ball.value().id())
+    {
+        // Do nothing if this robot is not the closest to the ball. Resolves issue of
+        // multiple simultaneous steals
+        return false;
+    }
+    if (nearest_enemy_to_ball.has_value())
+    {
+        // Get the ball if ball is closer to robot than enemy threat by threshold ratio
+        // and within max range
+        double ball_distance_to_friendly = distance(robot_position, ball_position);
+        double ball_distance_to_enemy =
+            distance(nearest_enemy_to_ball.value().position(), ball_position);
+
+        bool ball_is_near_friendly =
+            ball_distance_to_friendly <
+            ball_distance_to_enemy *
+                (1.0 - pass_defender_config.max_get_ball_ratio_threshold());
```

Comment:
this is kinda confusing. If the default is 0.3, `ball_distance_to_enemy * ...` becomes negative so it will never be true

Reply:
How so? Wouldn't it be essentially `ball_distance_to_enemy * 0.7`?